### PR TITLE
#21 add searched pl to db

### PR DESCRIPTION
**Pull request by Lajmeno**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/app/playlist/PlaylistController.java [hunk 1]
```diff
@@ -30,4 +33,16 @@
                         .map(playlistData -> PlaylistDTO.of(playlistData)));
     }
 
+    @PostMapping
+    public ResponseEntity savePlaylistForUser(@RequestBody PlaylistDTO playlist, Principal principal){
+        PlaylistData playlistData = PlaylistData.of(playlist);
+        playlistData.setSpotifyUserId(principal.getName());
+        if(playlistService.savePlaylist(playlistData).equals(Optional.empty())){
```
Comment (gossie) [inline backend/src/main/java/de/neuefische/app/playlist/PlaylistController.java]: Either write
```java
if(playlistService.savePlaylist(playlistData).isEmpty()) {
    return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
}
return ResponseEntity.status(HttpStatus.ACCEPTED).build();
```
or event better
```java
return ResponseEntity.of(playlistService.savePlaylist(playlistData))
```
The difference would be, that the second variant would return 200 or 404.

File: backend/src/main/java/de/neuefische/app/playlist/data/PlaylistTrackArtist.java [hunk 2]
```diff
@@ -12,4 +13,8 @@
     public static PlaylistTrackArtist of(SpotifyTracksArtists artists){
         return new PlaylistTrackArtist(artists.name());
     }
+
+    public static PlaylistTrackArtist of(PlaylistTrackArtistDTO artists){
```
Comment (gossie) [inline backend/src/main/java/de/neuefische/app/playlist/data/PlaylistTrackArtist.java]: Rename to artist?

File: frontend/src/PlaylistDisplay/PlaylistDetail.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ export default function PlaylistDetail(){
             }
             throw new Error("There is no Playlist with the requested id");
         })
-        .then(responseBody => {setPlaylist(responseBody); setReadyToRender("yes");})
+        .then(responseBody => {setPlaylist(responseBody); setReadyToRender(true);})
```
Comment (gossie) [inline frontend/src/PlaylistDisplay/PlaylistDetail.tsx]: If the function has multiple statements, use multiple lines:
```javascript
.then(responseBody => {
    setPlaylist(responseBody);
    setReadyToRender(true);
})
```

File: frontend/src/SpotifySearch/SearchPlaylistDetail.tsx [hunk 4]
```diff
@@ -26,14 +26,27 @@ export default function SearchPlaylistDetail(){
             }
             throw new Error("There is no Playlist with the requested id");
         })
-        .then(responseBody => {setPlaylist(responseBody); setReadyToRender("yes");})
+        .then(responseBody => {setPlaylist(responseBody); setReadyToRender(true); setErrorMessage("")})
```
Comment (gossie) [inline frontend/src/SpotifySearch/SearchPlaylistDetail.tsx]: same here

File: frontend/src/SpotifySearch/SpotifySearch.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ export default function SpotifySearch(){
             }
             throw new Error("Nothing found..");
          })
-        .then(responseBody => setPlaylists(responseBody))
+        .then(responseBody => {setPlaylists(responseBody); setErrorMessage("");})
```
Comment (gossie) [inline frontend/src/SpotifySearch/SpotifySearch.tsx]: and here

**Review by gossie — COMMENTED** [review]
Some remarks

File: backend/src/main/java/de/neuefische/app/playlist/PlaylistController.java (hunk 1 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/playlist/PlaylistController.java]: Ok I used the first variant

File: backend/src/main/java/de/neuefische/app/playlist/data/PlaylistTrackArtist.java (hunk 2 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/playlist/data/PlaylistTrackArtist.java]: Good point. I renamed it

File: frontend/src/PlaylistDisplay/PlaylistDetail.tsx (hunk 3 above)
Comment (Lajmeno) [reply frontend/src/PlaylistDisplay/PlaylistDetail.tsx]: I changed it

**Review by gossie — APPROVED** [review]

**Pull request merged**